app/scripts/models/brand.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+var Brand = (function () {
+
+  var Brand = function Brand(data) {
+    this.categoryIdsCsl = data.hasOwnProperty('categories') ? data.categories : null;
+    this.description = data.hasOwnProperty('description') ? data.description : null;
+    this.id = data.hasOwnProperty('id') ? parseInt(data.id) : null;
+    this.image = data.hasOwnProperty('image') ? data.image : null;
+    this.name = data.hasOwnProperty('name') ? data.name : null;
+    this.ranking = data.hasOwnProperty('ranking') ? data.ranking : null;
+    this.brandUrl = null; // @TODO: There does not seem to be a URL property in the data.
+
+    // Convert the category ids CSL to an array and remember it.
+    this.categoryIds = this.categoryIdsCsl === null ? [] : String(this.categoryIdsCsl).split(',');
+  };
+
+  Brand.prototype = {
+    getId: function getId() {
+      return this.id;
+    },
+
+    hasId: function hasId() {
+      return this.id !== null;
+    },
+
+    setId: function setId(newId) {
+      this.id = newId;
+    },
+
+    hasName: function hasName() {
+      return this.name !== null;
+    },
+
+    getName: function getName() {
+      return this.name;
+    },
+
+    setName: function setName(name) {
+      this.name = name;
+    },
+
+    hasImage: function hasImage() {
+      return this.image !== null;
+    },
+
+    getImage: function getImage() {
+      return this.image;
+    },
+
+    setImage: function setImage(imageData) {
+      this.image = imageData;
+    },
+
+    isImageAnUpload: function isImageAnUpload() {
+      // @TODO: This needs do more checks, really.
+      return Array.isArray(this.image) && this.image.length === 1;
+    },
+
+    getImageUploadFolder: function getImageUploadFolder() {
+      return 'Brand';
```

Comment:
Since this is a string, should we set this as a constant along with the others?

Reply:
Yes, you're right. I also realised it's not up to the models to know where the images get uploaded to, so I moved the information to the storage repository (and made them constants).